### ISSUE TITLE
magicbot: add feedback tunable

### ISSUE THREAD

**Issue by virtuald** [robotpy/robotpy-wpilib-utilities#22]
Example:

```
class Component:
    # this would automatically set an NT value at /components/component/angle
    @feedback
    def get_angle(self):
        return ... 
```

There's a few things I would like to solve:
- Need to provide sensor feedback in disabled mode -- would like to avoid adding extra user code to make that happen
- Would like to cache variables read at the beginning of each control loop, to avoid potential race conditions

**Comment by Twinters007** [comment]
Would the syntax need to be `get_xxx` and NT value be /components/component/xxx?

**Comment by virtuald** [comment]
I think the only thing that would need to be required is the `@feedback` decorator. I don't think I would want *every* get_ function to be added, otherwise you could accidentally trigger it.

**Comment by Twinters007** [comment]
I mean in order to get a key to put in nt, would you look for whatever comes after get_?

**Comment by virtuald** [comment]
Yeah, I think you could autodetect that. The feedback decorator would probably take an optional keyword parameter so that one could override it if they wanted.

**Comment by Twinters007** [comment]
Close?

**Comment by auscompgeek** [comment]
Fixed in #66. Closing.
